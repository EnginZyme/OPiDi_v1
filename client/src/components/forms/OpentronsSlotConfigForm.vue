<template>
  <div>
    <div class="slot_layout__grid">
      <DeckSlotCard :index="10" />
      <DeckSlotCard :index="11" />
      <div></div>
      <DeckSlotCard :index="7" />
      <DeckSlotCard :index="8" />
      <DeckSlotCard :index="9" />
      <DeckSlotCard :index="4" />
      <DeckSlotCard :index="5" />
      <DeckSlotCard :index="6" />
      <DeckSlotCard :index="1" />
      <DeckSlotCard :index="2" />
      <DeckSlotCard :index="3" />
    </div>
  </div>
</template>

<script>
import DeckSlotCard from "../base/DeckSlotCard";

/**
 * Displays all the Opentrons slot cards for user interaction.
 */
export default {
  name: "OpentronsSlotConfigForm",
  components: {
    DeckSlotCard,
  },
};
</script>

<style scoped>
.slot_layout__grid {
  display: grid;
  grid-row-gap: max(2vh, 10px);
  grid-column-gap: max(9px, 2vh);
  grid-template-columns: minmax(230px, 1fr) minmax(230px, 1fr) minmax(
      230px,
      1fr
    );
  grid-auto-rows: minmax(160px, 20vh);
  padding: 21px 160px 30px 160px;
}
</style>
