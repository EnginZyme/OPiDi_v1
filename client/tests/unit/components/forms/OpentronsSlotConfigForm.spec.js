import { shallowMount } from "@vue/test-utils";
import OpentronsSlotConfigForm from "@/components/forms/OpentronsSlotConfigForm";

describe("shallowMounted OpentronsSlotConfigForm", () => {
  const wrapper = shallowMount(OpentronsSlotConfigForm);

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
