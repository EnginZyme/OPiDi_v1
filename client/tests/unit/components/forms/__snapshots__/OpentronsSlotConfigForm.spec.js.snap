// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted OpentronsSlotConfigForm renders correctly 1`] = `
<div>
  <div
    class="slot_layout__grid"
  >
    <deckslotcard-stub
      index="10"
    />
     
    <deckslotcard-stub
      index="11"
    />
     
    <div />
     
    <deckslotcard-stub
      index="7"
    />
     
    <deckslotcard-stub
      index="8"
    />
     
    <deckslotcard-stub
      index="9"
    />
     
    <deckslotcard-stub
      index="4"
    />
     
    <deckslotcard-stub
      index="5"
    />
     
    <deckslotcard-stub
      index="6"
    />
     
    <deckslotcard-stub
      index="1"
    />
     
    <deckslotcard-stub
      index="2"
    />
     
    <deckslotcard-stub
      index="3"
    />
  </div>
</div>
`;
